python/GangaGaudi/Lib/Applications/GaudiBase.py

```diff
@@ -30,39 +30,40 @@ class GaudiBase(IPrepareApp):
     schema = {}
     docstr = 'The version of the application (like "v19r2")'
     schema['version'] = SimpleItem(preparable=1, defvalue=None,
-                                   typelist=['str', 'type(None)'], doc=docstr)
+                                   typelist=[str, None], doc=docstr)
     docstr = 'The platform the application is configured for (e.g. ' \
              '"slc4_ia32_gcc34")'
     schema['platform'] = SimpleItem(preparable=1, defvalue=None,
-                                    typelist=['str', 'type(None)'], doc=docstr)
+                                    typelist=[str, None], doc=docstr)
     docstr = 'The user path to be used. After assigning this'  \
              ' you can do j.application.getpack(\'Phys DaVinci v19r2\') to'  \
              ' check out into the new location. This variable is used to '  \
              'identify private user DLLs by parsing the output of "cmt '  \
              'show projects".'
     schema['user_release_area'] = SimpleItem(preparable=1, defvalue=None,
-                                             typelist=['str', 'type(None)'],
+                                             typelist=[str, None],
                                              doc=docstr)
     docstr = 'The name of the Gaudi application (e.g. "DaVinci", "Gauss"...)'
-    schema['appname'] = SimpleItem(preparable=1, defvalue=None, typelist=['str', 'type(None)'],
+    schema['appname'] = SimpleItem(preparable=1, defvalue=None, typelist=[str, None],
                                    hidden=1, doc=docstr)
     docstr = 'Location of shared resources. Presence of this attribute implies'\
              'the application has been prepared.'
-    schema['is_prepared'] = SimpleItem(defvalue=None,
+    schema['is_prepared'] = ComponentItem('shareddirs',
+                                       defvalue=None,
                                        strict_sequence=0,
                                        visitable=1,
                                        copyable=1,
-                                       typelist=['type(None)', 'str', ShareDir],
+                                       typelist=[None, str, ShareDir],
```

Comment:
I don't know but it was allowed to be so I'm not going to go chasing down whether it is ever set to be this or not. (I suspect not). and I'm not sure I agree that there is an underlying problem here?